### PR TITLE
[Feature](array-function) Support array_pushback function #17417

### LINKED ISSUE #17417: [Feature] (array function) support array_pushback function
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

_No response_

### Use case

_No response_

### Related issues

https://github.com/apache/doris/issues/17176

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liangjiawei1110**
# Proposed changes

Issue Number: close #17417 

## Problem summary
Implement array_pushback.
```mysql
mysql> select array_pushback([1, 2], 3);
+--------------------------------+
| array_pushback(ARRAY(1, 2), 3) |
+--------------------------------+
| [1, 2, 3]                      |
+--------------------------------+
1 row in set (0.01 sec)
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run p0

**Comment by liangjiawei1110** [comment]
run p0

**Comment by liangjiawei1110** [comment]
run p0

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run p0

**Comment by liangjiawei1110** [comment]
run build

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run p0

**Comment by liangjiawei1110** [comment]
run p1

**Comment by liangjiawei1110** [comment]
run buildall

File: be/src/vec/functions/array/function_array_pushback.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeNullable>(remove_nullable(arguments[0]));
+    };
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        auto src_column =
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
```
Comment (zclllyybb) [inline be/src/vec/functions/array/function_array_pushback.cpp:65]: dont use this, but`unpack_if_const`

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Comment by gitccl** [comment]
This logic is almost the same as `array_pushfront`, and I think we can use templates to perform the parts that differ between them.

**Comment by liangjiawei1110** [comment]
run buildall

File: be/src/vec/functions/array/function_array_pushback.cpp (hunk 1 above)
Comment (liangjiawei1110) [reply be/src/vec/functions/array/function_array_pushback.cpp]: fixed

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run p0

File: gensrc/script/doris_builtins_functions.py [hunk 2]
```diff
@@ -747,6 +747,25 @@
     [['array_pushfront'], 'ARRAY_VARCHAR',    ['ARRAY_VARCHAR', 'VARCHAR'], 'ALWAYS_NULLABLE'],
     [['array_pushfront'], 'ARRAY_STRING',     ['ARRAY_STRING', 'STRING'], 'ALWAYS_NULLABLE'],
 
+    [['array_pushback'], 'ARRAY_BOOLEAN',    ['ARRAY_BOOLEAN', 'BOOLEAN'], 'ALWAYS_NULLABLE'],
```
Comment (amorynan) [inline gensrc/script/doris_builtins_functions.py:750]: maybe use ARRAY_T is better ? think of it ?

Comment (liangjiawei1110) [reply gensrc/script/doris_builtins_functions.py:750]: But I think we should keep the same format with other array functions .

File: be/src/vec/functions/array/function_array_pushback.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeNullable>(remove_nullable(arguments[0]));
+    };
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        const auto& [src_column, src_const] =
+                unpack_if_const(block.get_by_position(arguments[0]).column);
```
Comment (zclllyybb) [inline be/src/vec/functions/array/function_array_pushback.cpp:66]: not the right way. maybe taking a look at https://selectdb.feishu.cn/docx/UtysdxKDkoJ5xoxtJEUcoN6ineU will help?

**Review by zclllyybb — CHANGES_REQUESTED** [review]

**Comment by liangjiawei1110** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by liangjiawei1110** [comment]
run p1

**Comment by amorynan** [comment]
LGTM

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run buildall

**Comment by liangjiawei1110** [comment]
run beut

**Comment by liangjiawei1110** [comment]
run beut

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**